src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,145 @@
+import duke.exception.DukeException;
+import duke.exception.ExtraDescriptionException;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidNumberException;
+
+/**
+ * Parser class.
+ */
+class Parser {
+    private static String[] oneLine;
+
+    /**
+     * check str is a number.
+     *
+     * @param str The str to be checked.
+     * @return Ture if str is a number, otherwise False.
+     */
+    private boolean isNumeric(String str) {
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    /**
+     * feature of find command.
+     *
+     * @return FindCommand object.
+     */
+    private FindCommand findFeature() throws DukeException {
+        if (oneLine.length == 1 || (oneLine.length == 2 && oneLine[1].isBlank())) {
+            throw new DukeException("The description of a " + oneLine[0].trim() + " cannot be empty.");
+        } else if (oneLine.length == 2 && !oneLine[1].isBlank()) {
+            return new FindCommand(oneLine);
+        } else {
+            throw new DukeException("error occur in findFeature in Parser");
+        }
+    }
+
+    /**
+     * feature of done command.
+     *
+     * @return DoneCommand object.
+     */
+    private DoneCommand doneFeature() throws DukeException {
+        if (oneLine.length == 1 || (oneLine.length == 2 && oneLine[1].isBlank())) {
+            throw new DukeException("The description of a " + oneLine[0].trim() + " cannot be empty.");
+        } else if (oneLine.length == 2 && !oneLine[1].isBlank()
+                && oneLine[1].trim().split(" ").length == 1 && isNumeric(oneLine[1].trim())) {
+            return new DoneCommand(oneLine);
+        } else if (oneLine.length == 2 && oneLine[1].trim().split(" ").length != 1) {
+            throw new ExtraDescriptionException("There is extra description for done");
+        } else {
+            throw new InvalidNumberException("the description should be a number");
+        }
+    }
+
+    /**
+     * feature of delete command.
+     *
+     * @return DeleteCommand object.
+     */
+    private DeleteCommand deleteFeature() throws DukeException {
+        if (oneLine.length == 1 || (oneLine.length == 2 && oneLine[1].isBlank())) {
+            throw new DukeException("The description of a " + oneLine[0].trim() + " cannot be empty.");
+        } else if (oneLine.length != 1 && !oneLine[1].isBlank()
+                && oneLine[1].trim().split(" ").length == 1 && isNumeric(oneLine[1].trim())) {
+            return new DeleteCommand(oneLine);
+        } else if (oneLine.length == 2 && oneLine[1].trim().split(" ").length != 1) {
+            throw new ExtraDescriptionException("There is extra description for delete");
+        } else {
+            throw new InvalidNumberException("the description should be a number");
+        }
+    }
+
+    /**
+     * feature of add command.
```

Comment:
From reading the code, I can't really make out what it is supposed to do. Perhaps more elaboration on comments and/or additional abstraction will help